java/src/enchcracker/EnchCrackerWindow.java

```diff
@@ -215,16 +238,36 @@ public void actionPerformed(ActionEvent arg0) {
 
 				if (bookshelves < 0 || bookshelves > 15) {
 					Log.info("Add info failed, bookshelf count invalid");
+					bookshelvesTextField.setBackground(new Color(1.0F,0.3F,0.0F));
 					return;
 				}
 
-				if (slot1 < 0 || slot1 > 30 || slot2 < 0 || slot2 > 30 || slot3 < 0 || slot3 > 30) {
-					Log.info("Add info failed, slot count invalid");
+				if(slot1<0||slot1>30){
+					Log.info("Add info failed, slot 1 count invalid");
+					slot1TextField.setBackground(new Color(1.0F,0.3F,0.0F));
+					return;
+				}
+				if(slot2<0||slot2>30){
+					Log.info("Add info failed, slot 2 count invalid");
+					slot2TextField.setBackground(new Color(1.0F,0.3F,0.0F));
+					return;
+				}
+				if(slot3<0||slot3>30){
+					Log.info("Add info failed, slot 3 count invalid");
+					slot3TextField.setBackground(new Color(1.0F,0.3F,0.0F));
 					return;
 				}
 
 				Log.info("Added info, b = " + bookshelves + ", s1 = " + slot1 + ", s2 = " + slot2 + ", s3 = " + slot3);
 
+				enchData.add(new EnchData(bookshelves,slot1,slot2,slot3));
```

Comment:
This would be better as `EnchData ench = new EnchData(bookshelves, slot1, slot2, slot3);` and then passing the same EnchData into both